src/main/java/com/redhat/cloud/notifications/db/BundleResources.java

```diff
@@ -2,145 +2,79 @@
-
-        return connectionPublisher.get().onItem()
-            .transformToMulti(c -> Multi.createFrom().resource(() -> c,
-                c2 -> {
-                    Flux<PostgresqlResult> execute = c2.createStatement(query)
-                            .bind("$1", bundleId)
-                            .execute();
-
-                    return execute.flatMap(res -> res.map((row, rowMetadata) -> {
-                        Application application = new Application();
-                        application.setId(row.get("id", UUID.class));
-                        application.setName(row.get("name", String.class));
-                        application.setDisplay_name(row.get("display_name", String.class));
-                        application.setBundleId(bundleId);
-                        return application;
-                    }));
-                })
-                .withFinalizer(postgresqlConnection -> {
-                    postgresqlConnection.close().subscribe();
-                }));
+    public Multi<Application> getApplications(UUID id) {
+        String query = "FROM Application WHERE bundle.id = :id";
+        return session.createQuery(query, Application.class)
+                .setParameter("id", id)
+                .getResultList()
+                .onItem().transformToMulti(Multi.createFrom()::iterable);
     }
 
     public Uni<Application> addApplicationToBundle(UUID bundleId, Application app) {
-        String insertQuery = "INSERT INTO " + PUBLIC_APPLICATIONS + " (name, display_name, bundle_Id) VALUES ($1, $2, $3)";
-
-        return connectionPublisher.get().onItem()
-                .transformToMulti(c -> Multi.createFrom().resource(() -> c,
-                    c2 ->  c2.createStatement(insertQuery)
-                            .bind("$1", app.getName())
-                            .bind("$2", app.getDisplay_name())
-                            .bind("$3", bundleId)
-                            .returnGeneratedValues("id")
-                            .execute()
-                            .flatMap(res -> res.map((row, rowMetadata) -> {
-                                app.setId(row.get("id", UUID.class));
-                                return app;
-                            }))
-
-                )
-                        .withFinalizer(postgresqlConnection -> {
-                            postgresqlConnection.close().subscribe();
-                        }))
-                .toUni();
+        return Uni.createFrom().item(app)
+                .onItem().transform(a -> addBundleReference(a, bundleId))
+                .onItem().transformToUni(session::persist)
+                .call(session::flush)
+                .replaceWith(app);
     }
 
-
+    /**
+     * Adds to the given {@link Application} a reference to a persistent {@link Bundle} without actually loading its
+     * state from the database. The app will remain unchanged if the given bundle identifier is null.
```

Comment:
What would be a use-case for a null bundle? Should we perhaps throw an IllegalArgumentException?

Reply:
I wanted this method to be null-proof and considered validation should be performed at an earlier stage, but it might be overkill. I checked the code, `bundleId` can't be `null` with the current version.